app/models/claim_state_transition_reason.rb

```diff
@@ -72,22 +152,29 @@ def refuse_reasons_for(claim)
       reasons_for("refused_#{claim.class.to_s.demodulize.tableize}")
     end
 
+    def transition_reasons
```

Comment:
**[not a blocker]** Have you considered having this to return the list of `ClaimStateTransitionReason` object instances instead of the so that methods below like `reasons_for(state)` just return the matching instance, rather done initialising one.